flexmeasures/data/models/reporting/__init__.py

```diff
@@ -1,123 +1,42 @@
 from __future__ import annotations
-from typing import Optional, Union, Dict
-from datetime import datetime, timedelta
 
-import pandas as pd
-
-from flexmeasures.data.schemas.reporting import ReporterConfigSchema
 from flexmeasures.data.models.time_series import Sensor
-from flexmeasures.data.models.data_sources import DataGeneratorMixin
+from flexmeasures.data.models.data_sources import DataGenerator
+
+from flexmeasures.data.schemas.reporting import (
+    ReporterInputsSchema,
+    ReporterConfigSchema,
+)
 
 import timely_beliefs as tb
 
 
-class Reporter(DataGeneratorMixin):
+class Reporter(DataGenerator):
     """Superclass for all FlexMeasures Reporters."""
 
     __version__ = None
     __author__ = None
-    __data_generator_base__ = "Reporter"
+    __data_generator_base__ = "reporter"
 
     sensor: Sensor = None
 
-    reporter_config: Optional[dict] = None
-    reporter_config_raw: Optional[dict] = None
-    schema = ReporterConfigSchema()
-    data: Dict[str, Union[tb.BeliefsDataFrame, pd.DataFrame]] = None
-
-    def __init__(
-        self, sensor: Sensor, reporter_config_raw: Optional[dict] = None
-    ) -> None:
-        """
-        Initialize a new Reporter.
-
-        Attributes:
-        :param sensor: sensor where the output of the reporter will be saved to.
-        :param reporter_config_raw: dictionary with the serialized configuration of the reporter.
-        """
-
-        self.sensor = sensor
-
-        if not reporter_config_raw:
-            reporter_config_raw = {}
+    _input_schema = ReporterInputsSchema()
+    _config_schema = ReporterConfigSchema()
 
-        self.reporter_config_raw = reporter_config_raw
-
-    def fetch_data(
-        self,
-        start: datetime,
-        end: datetime,
-        input_resolution: timedelta = None,
-        belief_time: datetime = None,
-    ):
-        """
-        Fetches the time_beliefs from the database
-        """
+    def __init__(self, config: dict | None = None, **kwargs) -> None:
+        super().__init__(config, **kwargs)
```

Comment:
Looks like this doesn't do anything anymore now, and could go.

Reply:
True